Framework/src/ServiceDiscovery.cxx

```diff
@@ -96,6 +97,11 @@ bool ServiceDiscovery::_register(const std::string& objects)
   check.put("Name", "Health check " + mId);
   check.put("Interval", "5s");
   check.put("DeregisterCriticalServiceAfter", "1m");
+  // Wait until port is set by the thread
+  {
+    std::unique_lock<std::mutex> lk(mHealthPortMutex);
+    mHealthPortCV.wait(lk, [this] { return mHealthPortAssigned == true; });
```

Comment:
Maybe I misunderstood how `condition_variable` works, but won't it block the second time you call `_register`? I am not sure if it requires the other thread to call `notify_one()` for each `wait`.

Reply:
OK, I wrote a little reproducer and it seems it my concern is not valid, it seems that `notify_one()` unblocks the main thread forever, as long as the predicate is provided in `wait`.

Reply:
as you noted, as long as the predicate is true, it will let the main thread continue.